### PR TITLE
Misc XAlloc cleanup

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: XAlloc.c [hunk 1]
```diff
@@ -1,41 +1,48 @@
-
 #include "XAlloc.h"
-#include "RichString.h"
 
+#include "CRT.h"
+
+/* The function vasprintf is a GNU extension. */
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
```
Comment (BenBE) [inline XAlloc.c:8]: Isn't this set by `configure`/the `Makefile`?

Comment (cgzones) [reply XAlloc.c]: Indeed; replaced with a`#include "config.h"`

**Comment by cgzones** [comment]
Scheduled after #160 and #161

**Review by BenBE — APPROVED** [review]
LGTM.

**Pull request merged**